tests/files/fips-test-gnutls.c

```diff
@@ -0,0 +1,63 @@
+/** Simple C program that will calculate the hash of a fixed string using
+ * GnuTLS.
+ *
+ * The hash to be used is passed as the first parameter to the binary.
+ * This program demonstrates the use of GnuTLS for computing message digests.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <gnutls/gnutls.h>
+#include <gnutls/crypto.h>
+#include <stdlib.h>
+#include <assert.h>
+
+int main(int argc, char *argv[]) {
+    gnutls_digest_algorithm_t digest_algorithm;
+    const char *mess1 = "Test Message\n";
+    const char *mess2 = "Hello World\n";
+    unsigned char md_value[128];
+    size_t md_len;
+
+    if (argc < 2) {
+        fprintf(stderr, "Usage: mdtest digestname\n");
+        return 1;
+    }
+
+    gnutls_global_init();
+    digest_algorithm = gnutls_digest_get_id(argv[1]);
+    if (digest_algorithm == GNUTLS_DIG_UNKNOWN) {
+        fprintf(stderr, "Unknown message digest %s\n", argv[1]);
+        gnutls_global_deinit();
+        return 1;
```

Comment:
you could have a section like this at the end:
```C
cleanup:
    gnutls_global_deinit();
    return 1;
```
and instead do a 
```suggestion
        goto cleanup;
```